itests/src/test/java/fi/metatavu/edelphi/test/mock/QueryMocker.java

```diff
@@ -0,0 +1,272 @@
+      Date closes, String state) {
+    long id = getNextId("Resource");
+    createQuery(id, name, urlName, parentFolderId, visible, creatorId, created, modifierId, modified, description,
+        indexNumber, allowEditReply, closes, state);
+    queryIds.add(id);
+    return id;
+  }
+
+  private void createQuery(Long id, String name, String urlName, Long parentFolderId, boolean visible, Long creatorId,
+      Date created, Long modifierId, Date modified, String description, Integer indexNumber, boolean allowEditReply,
+      Date closes, String state) {
+    String resourceSql = "INSERT INTO "
+        + "  `resource` (`id`, `name`, `urlName`, `parentFolder_id`, `type`, `visible`, `archived`, `creator_id`, `created`, `lastModifier_id`, `lastModified`, `description`, `indexNumber`) "
+        + "VALUES " + "  (?,?,?,?,?,?,?,?,?,?,?,?,?)";
+
+    String querySql = "INSERT INTO " + "  `query` (`id`, `allowEditReply`, `closes`, `state`) " + "VALUES "
+        + "  (?,?,?,?)";
+
+    executeSql(resourceSql, id, name, urlName, parentFolderId, "QUERY", visible, false, creatorId, created, modifierId,
+        modified, description, indexNumber);
+    executeSql(querySql, id, allowEditReply, closes, state);
+  }
+
+  private Long createQuerySection(Long queryId, String title, Long creatorId, Date created, Long modifierId,
+      Date modified, boolean visible, Integer sectionNumber, boolean commentable, boolean viewDiscussions) {
+    long id = getNextId("QuerySection");
+    createQuerySection(id, queryId, title, creatorId, created, modifierId, modified, visible, sectionNumber,
+        commentable, viewDiscussions);
+    sectionIds.add(id);
+    return id;
+  }
+
+  private void createQuerySection(Long id, Long queryId, String title, Long creatorId, Date created, Long modifierId,
+      Date modified, boolean visible, Integer sectionNumber, boolean commentable, boolean viewDiscussions) {
+    String sql = "INSERT INTO "
+        + "  `querysection` (`id`, `query_id`, `title`, `creator_id`, `created`, `lastModifier_id`, `lastModified`, `visible`, `archived`, `sectionNumber`, `commentable`, `viewDiscussions`) "
+        + "VALUES " + "  (?,?,?,?,?,?,?,?,?,?,?,?)";
+
+    executeSql(sql, id, queryId, title, creatorId, created, modifierId, modified, visible, false, sectionNumber,
+        commentable, viewDiscussions);
+  }
+
+  private long createQueryPage(Integer pageNumber, Long queryId, Long creatorId, Date created, Long modifierId, Date modified,
+      String type, String title, Long sectionId, boolean visible) {
+    long id = getNextId("QueryPage");
+    createQueryPage(id, pageNumber, creatorId, created, modifierId, modified, type, title, sectionId, visible);
+    pages.add(new PageMeta(id, queryId, sectionId, type));
+    return id;
+  }
+
+  private void createQueryPage(Long id, Integer pageNumber, Long creatorId, Date created, Long modifierId,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Method has 10 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00107)